src/Razor/src/Microsoft.NET.Sdk.Razor/ReferenceResolver.cs

```diff
@@ -46,11 +49,17 @@ public IReadOnlyList<string> ResolveAssemblies()
             return applicationParts;
         }
 
-        private DependencyClassification Resolve(ClassifiedAssemblyItem classifiedItem)
+        private void Resolve(ClassifiedAssemblyItem classifiedItem, HashSet<string> visited)
         {
             if (classifiedItem.DependencyClassification != DependencyClassification.Unknown)
             {
-                return classifiedItem.DependencyClassification;
+                return;
+            }
+
+            if (!visited.Add(classifiedItem.Name))
```

Comment:
We only ever do something with the returned value if it's `ReferencesMvc` or is an Mvc assembly. (I should probably rename these enums since the code fairly general purpose now). Returning this with the value being `Unknown` is 🆗